test/test_common/environment.cc

```diff
@@ -206,4 +211,18 @@ void TestEnvironment::exec(const std::vector<std::string>& args) {
     RELEASE_ASSERT(false);
   }
 }
+
+std::string TestEnvironment::writeStringToFileForTest(const std::string& filename,
+                                                      const std::string& contents) {
+  // We rather unfortunately have to use runfilesPath becauase many consumers of
+  // these files use temporaryFileSubstitute, which assumes runfilesPath.
+  const std::string out_path = TestEnvironment::runfilesPath(filename);
+  RELEASE_ASSERT(::system(("mkdir -p $(dirname " + out_path + ")").c_str()) == 0);
```

Comment:
Oh that was lazy-copied from temporaryFileSubstitute
It's not necessary if you only do filenames with no directory paths which I think is a reasonable thing to do especially as this is test code and will fast-fail if you add a new test which does it wrong.